sbi/utils/sbiutils.py

```diff
@@ -224,7 +229,8 @@ def sample_posterior_within_prior(
 
     pbar.close()
 
-    return torch.cat(accepted), as_tensor(acceptance_rate)
+    # When in case of leakage a batch size was used there could be too many samples.
+    return torch.cat(accepted)[:num_samples], as_tensor(acceptance_rate)
```

Comment:
Maybe we should add an assertion here? Sth like

```
samples = torch.cat(accepted)[:num_samples]
assert samples.shape[0] == num_samples
return samples, as_tensor(acceptance_rate)
```

Reply:
The while-loop above is written such that it returns only when `num_remaining<=0` which iff `samples.shape[0]>=num_samples`, no?

But if you prefer we could still add an assertion to double check.

Reply:
True, should suffice in principle but probably won't hurt? Should be super quick in terms of evaluation costs